### ISSUE TITLE
Check for updates doesn't seem to work anymore

### ISSUE THREAD

**Issue by Janitsu** [ModOrganizer2/modorganizer#1554]
## The problem:
Clicking the _Check for updates_ button doesn't seem to be checking the mods for updates for some reason.

## To Reproduce:

1. Open Mod Organiser 2
2. Click on _Check for updates_
3. No new updates noticed by the Mod Organiser 2 but manually checking reveals that there are updates
4. Sometimes it seems that force-checking them works(?)

## Environment:

* Mod Organizer Version that exhibits the issue: 2.4.2
* Last Mod Organizer Version that did not exhibit the issue (if applicable):
* Desktop OS/version used to run Mod Organizer: Windows 10 Home

## Details:
Whenever clicking the _Check for updates_ in the drop-down menu as indicated in the picture below, the log says that it has checked all the files for updates and there are seemingly no updates for any of the mods.

![image](https://user-images.githubusercontent.com/8763688/125212394-c7901a80-e2b5-11eb-9d7b-427ecde08b55.png)

However, I grew suspicious of this behaviour of modders suddenly not updating their mods at all, considering that there were updates almost daily at one point, I went into download history on Nexusmods and noticed that a few of them had been updated recently with different version numbers. These weren't shown on Mod Organiser 2 and the log said that nothing is out of order.

![image](https://user-images.githubusercontent.com/8763688/125212416-eee6e780-e2b5-11eb-9c61-5b83c4293967.png)

Here in the log, it says no mods have been updated on Nexus. When I downloaded a new version of one of the mods that I have installed, it suddenly got the notification for having a newer version on Nexusmods.

![image](https://user-images.githubusercontent.com/8763688/125212437-0f16a680-e2b6-11eb-8c64-64a1b9c8e543.png)

## Link to Mod Organizer logs:
Right click on the log window at the bottom of MO2, set log level to debug, restart Mo2 and reproduce the problem.
To access the log files you can press the Open folder icon over the modlist and select "Open MO2 Logs folder".

### USVFS:
Create a [GIST](https://gist.github.com) which is a paste of your latest USVFS log, and link them here.
Do _NOT_ paste your full USVFS logs here, as it will make this issue very long and hard to read!
If you are reporting a bug, _always_ include the USVFS logs!
https://gist.github.com/Janitsu/4c88091dba71820c4b47a3b72c1341d2

### MO Interface:
Create a [GIST](https://gist.github.com) which is a paste of your mo_interface logs, and link them here.
Do _NOT_ paste your full mo_interface logs here, as it will make this issue very long and hard to read!
If you are reporting a bug, _always_ include the mo_interface logs!
https://gist.github.com/Janitsu/afc4478a2a71aed4ddeecf605c399e06

**Comment by Al12rs** [comment]
Spent all today looking into it. I published a test build on Discord https://discord.gg/6GKR9jZ
Let me know if it works fine

**Comment by Janitsu** [comment]
Installed the test build and it immediately seems to pick up on the mods. I will be double-checking with download history to see how accurate it is at doing that over the week or so. Thanks!

**Comment by Al12rs** [comment]
Thanks for testing it! I'm not too confident I got everything right so it's better to keep an eye on it

**Comment by Al12rs** [comment]
Closed with #1555

**Comment by Janitsu** [comment]
Sorry for commenting again on this same post, but I think it falls under the same category.

At first, the update checking didn't really do anything differently from the non-dev version but I kept on using it nevertheless to see if there is any change in behaviour. I deleted the Nexus cache and now it seems to be checking the mods perfectly well. One little caveat I noticed is that whenever I use the "Check for updates" from the menu, it tells me that none of my mods are in need of an update only for the mods that require an update to show up on the list in the following few seconds.

I will continue to monitor the update checking as well as keeping an eye on the cache and whether cleaning the files helps or not. Thank you for the update! 👍

**Comment by Al12rs** [comment]
Maybe it says "None of your **Skyrim** mods appear to have had recent file updates." while you are modding SSE? Mo2 asks nexus for updates in the last X days for each Nexus game that can be source of your mods. So for SSE it has to check both Skyrim and Skyrimspecialedition.

**Comment by Janitsu** [comment]
Ah, I should've included a screenshot or a log file now that I think of it. The log says that neither Skyrim nor SSE mods would be out of date. I'll try and wait to get some examples (unless it was a unique incident). If it was only a single exception to the rule due to the updated MO2 or deleting the cache files, it wouldn't even be worth mentioning. Thanks for the quick reply!

**Comment by Janitsu** [comment]
I think I've managed to track down what is causing the issue for me. I noticed that the webcache wasn't able to delete a certain file and it seems to bug out the update check. If I check for updates without doing anything else than launching MO2, it will not notify me of any updates. However, if I clear cache in "Nexus" menu in settings, it seems to be able to check for mod updates as intended.

![image](https://user-images.githubusercontent.com/8763688/126372247-a99143f0-4a9f-46e9-8c8b-7a2df5e5b6f7.png)
